include/swift/AST/DiagnosticsSema.def

```diff
@@ -2693,6 +2693,10 @@ ERROR(differentiable_attr_no_parameters,none,
       "%0 has no parameters to differentiate with respect to", (DeclName))
 ERROR(differentiable_attr_void_result,none,
       "cannot differentiate void function %0", (DeclName))
+ERROR(differentiable_attr_primal_no_definition,none,
+      "cannot specify primal on function declaration without definition", ())
```

Comment:
The only case of functions not having a definition in Swift is protocol requirements :) So we should call it "protocol requirements" instead of "function declaration without definition" like other diagnostics.